packages/bitcore-lib/lib/publickey.js

```diff
@@ -281,6 +296,115 @@ PublicKey.fromX = function(odd, x) {
   });
 };
 
+/**
+ * PublicKey instance from a Taproot (32-byte) public key
+ * @param {String|Buffer} hexBuf 
+ * @returns {PublicKey}
+ */
+PublicKey.fromTaproot = function(hexBuf) {
+  if (typeof hexBuf === 'string' && JSUtil.isHexaString(hexBuf)) {
+    hexBuf = Buffer.from(hexBuf, 'hex');
+  }
+  $.checkArgument(Buffer.isBuffer(hexBuf), 'input must be a hex string or buffer');
+  $.checkArgument(hexBuf.length === 32, 'Taproot public keys must be 32 bytes');
+  return new PublicKey.fromX(false, hexBuf);
+}
+
+/**
+ * Verifies if the input is a valid Taproot public key
+ * @param {String|Buffer} hexBuf 
+ * @returns {Boolean}
+ */
+PublicKey.isValidTaproot = function(hexBuf) {
+  if (typeof hexBuf === 'string' && JSUtil.isHexaString(hexBuf)) {
+    hexBuf = Buffer.from(hexBuf, 'hex');
+  }
+  $.checkArgument(Buffer.isBuffer(hexBuf), 'input must be a hex string or buffer');
+  $.checkArgument(hexBuf.length === 32, 'Taproot public keys must be 32 bytes');
+
+  // TODO: do a more thorough taproot validation
+
+  // if (!secp256k1_fe_set_b32(&x, input32)) {
+  //   return false;
+  // }
+  // if (!secp256k1_ge_set_xo_var(&pk, &x, 0)) {
+  //     return false;
+  // }
+  // if (!secp256k1_ge_is_in_correct_subgroup(&pk)) {
+  //     return false;
+  // }
+  return true;
+};
+
+
+/**
+ * Get the TapTweak tagged hash of this pub key and the merkleRoot
+ * @param {Buffer} merkleRoot (optional)
+ * @returns {Buffer}
+ */
+PublicKey.prototype.computeTapTweakHash = function(merkleRoot) {
+  const taggedWriter = new TaggedHash('TapTweak');
+  taggedWriter.write(this.point.x.toBuffer({ size: 32 }));
+
+  //  If !merkleRoot, then we have no scripts. The actual tweak does not matter, but 
+  //  follow BIP341 here to allow for reproducible tweaking.
+
+  if (merkleRoot) {
+    $.checkArgument(Buffer.isBuffer(merkleRoot) && merkleRoot.length === 32, 'merkleRoot must be 32 byte buffer');
+    taggedWriter.write(merkleRoot);
+  }
+  const tweakHash = taggedWriter.finalize();
+  
+  const order = Point.getN();
+  $.checkState(BN.fromBuffer(tweakHash).lt(order), 'TapTweak hash failed secp256k1 order check');
+  return tweakHash;
+};
+
+
+/**
+ * Verify a tweaked public key against this key
+ * @param {PublicKey|Buffer} p Tweaked pub key
+ * @param {Buffer} merkleRoot (optional)
+ * @param {Buffer} control 
+ * @returns {Boolean}
+ */
+PublicKey.prototype.checkTapTweak = function(p, merkleRoot, control) {
+  if (Buffer.isBuffer(p)) {
+    p = PublicKey.fromTaproot(p);
+  }
+  const tweak = p.computeTapTweakHash(merkleRoot);
+
+  const P = p.point.liftX();
+  const Q = P.add(this.point.curve.g.mul(BN.fromBuffer(tweak)));
+  
+  if (!this.point.x.eq(Q.x)) {
```

Comment:
Instead of 380 - 386, you could:
`return this.point.x.eq(Q.x) && Q.y.mod(new BN(2)).eq(new BN(control[0] & 1)`